x/gamm/pool-models/stableswap/pool.go

```diff
@@ -299,15 +315,16 @@ func (p *Pool) PokePool(blockTime time.Time) {}
 // It should only be able to be successfully called by the pool's ScalingFactorGovernor
 // TODO: move commented test for this function from x/gamm/keeper/pool_service_test.go once a pool_test.go file has been created for stableswap
 func (p *Pool) SetStableSwapScalingFactors(ctx sdk.Context, scalingFactors []uint64, scalingFactorGovernor string) error {
-	if scalingFactorGovernor != p.ScalingFactorGovernor {
+	if scalingFactorGovernor != p.ScalingFactorGovernor || len(scalingFactors) != p.PoolLiquidity.Len() {
 		return types.ErrNotScalingFactorGovernor
 	}
```

Comment:
Err should be a different error message and therefore a second check

Reply:
Check moved into `validateScalingFactors` so this should be addressed